docs/software-platforms/docker/docker-images.md

```diff
@@ -0,0 +1,114 @@
+# Creating Docker images
+
+This page provides some guidance on creating Docker images for an OpenScPCA analysis module.
+
+Note that creating docker images is not required for contributing to OpenScPCA, but it can be useful for testing and sharing your analysis module with others.
+Each analysis module will ultimately be updated to include a Docker image to allow for running the analysis module as part of the `OpenScPCA-nf` workflow, but this does not need to be done by the module author.
+Data Lab staff will create a Docker image as part of the process of adding the module to the `OpenScPCA-nf` workflow if it has not already been done.
+
+## Prerequisites
+
+Before making a Docker image, you will likely want to have fully [defined the software environment](../../contributing-to-analyses/determining-requirements/determining-software-requirements.md) that you will be using for your analysis module.
+Most likely, this will take the form of either:
+
+- an [`renv.lock` file](../../contributing-to-analyses/determining-requirements/determining-software-requirements.md#determining-and-managing-software-dependencies-in-r) if you are working with R, or
+- a [conda `environment.yml`](../../contributing-to-analyses/determining-requirements/determining-software-requirements.md#module-specific-conda-environments) and/or [`conda-lock.yml` file](../../contributing-to-analyses/determining-requirements/determining-software-requirements.md#freezing-dependencies-with-conda-lock) if you are working with Python or other software available through [Bioconda](https://bioconda.github.io).
+
+The goal of creating a Docker image is to incorporate these software requirements into a container that can be used to run all parts of the analysis module without any installation steps other than installing Docker and pulling the image.
+
+You will also need to have [Docker installed on your computer](https://docs.docker.com/engine/install/) to build and test the images.
+
+## Docker template files
+
+When you create an analysis module, a template `Dockerfile` and a `.dockerignore` file will be created in the root of the analysis module.
+
+The  `Dockerfile` includes a `FROM` statement that specifies the base image to use and a couple of `LABEL` statements to populate metadata about the image.
+
+The `.dockerignore` file is used to specify files and directories that should *not* be included in the Docker build environment.
+By default, this will be set to ignore all files with the exception of environment files (e.g., `renv.lock`, `environment.yml`, `conda-lock.yml`).
```

Comment:
```suggestion
By default, this will be set to ignore all files except environment files (e.g., `renv.lock`, `environment.yml`, `conda-lock.yml`).
```